.github/workflows/CI.yml

```diff
@@ -1,31 +1,35 @@
 # This workflow will install Python dependencies, run tests and lint with a variety of Python versions
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
 
-name: CI_version-2.02
+name: CI
 
 on:
   push:
     branches: [version-2.02]
   pull_request:
     branches: [version-2.02]
 
+concurrency: CI-2.02
+
 jobs:
   build:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: ["3.7"]
 
     steps:
-      - name: Install xmllint
-        run: sudo apt-get install libxml2-utils
-
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+      - name: Set up Python
+        uses: actions/setup-python@v4
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: '3.9'
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get -y install libxml2-dev libxslt-dev libxslt1-dev python3-dev python3-lxml libxml2-utils
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install -r requirements.txt
 
       - uses: actions/cache@v2
```

Comment:
Should be able to remove this and use the built-in feature of `actions/setup-python` 
https://github.com/actions/setup-python#caching-packages-dependencies

Reply:
I can move the pip parts into setup-python, but the libxml and libxslt dependencies will need to be apt-get installed prior to it. Without those trying to pip install lxml always fails.

Reply:
Ohh shoot my comment got added in the wrong place. I meant it to be on the `- uses: actions/cache@v2` line!

Reply:
Okay good point. So far IATI-Codelists was the only repo with a relatively robust GH actions yml, so I carried forward all the previous caching. But you're right we can simplify it by moving the cache from actions/cache to setup-python cache.